### PR TITLE
Move to pyproject

### PR DESCRIPTION
**Pull request by demianbrecht**
- [ ] Update CHANGELOG.md
- [ ] Update README.md (as needed)
- [ ] New dependencies were added to `requirements.txt` or `requirements-dev.txt`
- [ ] `pip install` succeeds with a clean virtualenv
- [ ] There are new or modified tests that cover changes
- [ ] Test coverage is maintained or expanded

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -9,74 +9,34 @@ on:
...
-        run: pip install wheel
+        run: python -m pip install --upgrade build
 
-      - name: build sdist
-        run: python setup.py sdist
+      - name: build
+        run: python -m build
 
-      - name: cache buildfiles
-        uses: actions/cache@v2
```
[the remark this answers is not in the document]
Comment (bgrant) [reply .github/workflows/publish-release.yml:27]: Nevermind - you deleted these lines.

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,54 @@
...
+  "pyyaml~=6.0",
+  "sphinx_rtd_theme==0.5.2",
+  "tblib~=1.6.0",
+]
+[tool.setuptools.dynamic]
+readme = {file = ["README.md"]}
+
 [tool.black]
 line-length = 88
 target_version = ['py37']
```
Comment (bgrant) [inline pyproject.toml:54]: I guess we should change the black `target_version` to `'py39'` too

**Comment by bgrant** [comment]
Can you add a CHANGELOG entry?

File: .github/workflows/publish-release.yml [hunk 3]
```diff
@@ -9,74 +9,34 @@ on:
     types: [created]
 
 jobs:
-  build-source:
+  build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
```
[the remark this answers is not in the document]
Comment (demianbrecht) [reply .github/workflows/publish-release.yml:16]: Yeah, out of scope for this imo

File: .bumpversion.cfg [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [bumpversion]
-current_version = 0.24.0
+current_version = 0.24.1
```
Comment (bgrant) [inline .bumpversion.cfg:2]: Usually we wait until the release PR to bump the version, but I don't mind if you're planning on release right away for testing anyway.

Comment (demianbrecht) [reply .bumpversion.cfg]: I'll hold off

**Pull request merged**